### PR TITLE
[New Rule] Potential Remote Desktop Shadowing Activity

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/1100

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/lateral_movement_evasion_rdp_shadowing.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+creation_date = "2021/04/12"
+maturity = "production"
+updated_date = "2021/04/12"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the modification of the Remote Desktop Protocol (RDP) Shadow registry or the execution of processes
+indicative of active RDP shadowing session. An Adversary may abuse RDP Shadowing feature to spy on other users active
+RDP sessions or even control them.
```
Comment (bm11100) [inline rules/windows/lateral_movement_evasion_rdp_shadowing.toml:11]:
```suggestion
Identifies the modification of the Remote Desktop Protocol (RDP) Shadow registry or the execution of processes
indicative of an active RDP shadowing session. An adversary may abuse the RDP Shadowing feature to spy on or control other users active
RDP sessions.
```

**Review by bm11100 — APPROVED** [review]

File: rules/windows/lateral_movement_evasion_rdp_shadowing.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+/* Identifies the modification of RDP Shadow registry or
+  the execution of processes indicative of active shadow RDP session */
+any where event.category : ("registry", "process") and
+  registry.path : "HKLM\\Software\\Policies\\Microsoft\\Windows NT\\Terminal Services\\Shadow" or
+  (process.name : ("RdpSaUacHelper.exe", "RdpSaProxy.exe") and process.parent.name : "svchost.exe") or
+  (process.pe.original_file_name : "mstsc.exe" and process.args : "/shadow:*")
```
Comment (brokensound77) [inline rules/windows/lateral_movement_evasion_rdp_shadowing.toml:34]: I don't think we need `registry where process.name : ("RdpSaUacHelper.exe", "RdpSaProxy.exe")` right? In which case, this logic is a little tighter and easier to read

```suggestion
/* Identifies the modification of RDP Shadow registry or
  the execution of processes indicative of active shadow RDP session */

any where 
  (event.category == "registry" and
     registry.path : "HKLM\\Software\\Policies\\Microsoft\\Windows NT\\Terminal Services\\Shadow"
  ) or
  (event.category == "process" and 
     (process.name : ("RdpSaUacHelper.exe", "RdpSaProxy.exe") and process.parent.name : "svchost.exe") or
     (process.pe.original_file_name : "mstsc.exe" and process.args : "/shadow:*")
  )
```

Comment (Samirbous) [reply rules/windows/lateral_movement_evasion_rdp_shadowing.toml:34]: it's not needed cuz the execution of "RdpSaUacHelper.exe" or "RdpSaProxy.exe catches the run-time state (registry is supposed to be set before/existing shadow rdp) and in itself is enough to alert. registry change is creation-state detection.

Comment (brokensound77) [reply rules/windows/lateral_movement_evasion_rdp_shadowing.toml]: 👍 just making sure

**Review by brokensound77 — APPROVED** [review]
🚢

**Pull request merged**